packages/vscode/src/fill/zip.ts

```diff
@@ -58,81 +58,38 @@ export function zip(tarPath: string, files: IFile[]): Promise<string> {
 	});
 }
 
-export async function extract(tarPath: string, targetPath: string, options: IExtractOptions = {}, token: CancellationToken): Promise<void> {
-	const sourcePathRegex = new RegExp(options.sourcePath ? `^${options.sourcePath}` : '');
-
+export async function extract(archivePath: string, extractPath: string, options: IExtractOptions = {}, token: CancellationToken): Promise<void> {
 	return new Promise<void>(async (c, e) => {
-		const buffer = await promisify(fs.readFile)(tarPath);
-		const extractor = tarStream.extract();
-		extractor.once('error', e);
-		extractor.on('entry', (header, stream, next) => {
-			const rawName = header.name;
+		extractZip(archivePath, extractPath, options, token).then(c).catch((ex) => {
```

Comment:
Should we try extracting a tar first? We expect _most_ extensions to be in the tar format.

Reply:
That makes sense, extraction only occurs for installing from the marketplace.